src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -367,6 +367,15 @@ protected virtual async Task ProcessHeadersAsync(INetworkPeer peer, List<BlockHe
                 {
                     this.cachedHeaders.AddRange(headers);
                     this.logger.LogDebug("All {0} items were not consumed and added to cache.", headers.Count);
+                    this.logger.LogTrace("(-)[HEADERS_NOT_CONSUMED]");
+
+                    return;
+                }
+
+                if (result.Consumed == this.BestReceivedTip)
+                {
+                    this.logger.LogDebug("No new headers where presented");
```

Comment:
`were`

Reply:
Not to steal your thunder but I'm all about this new feature atm

```suggestion
                    this.logger.LogDebug("No new headers were presented");
```